### PR TITLE
Jack mix box updates

### PR DESCRIPTION
**Pull request by synthnassizer**
- added cli option to get jack_mix_box to work in "volume pickup" mode,
- minor typo fix in cli help

### CODE REVIEW

**Comment by SpotlightKid** [comment]
I made a small commit to `jack_mix_box.c`  in the same place this afternoon but forgot to push it to this repo, which I did now. Can you please rebase your PR?

**Comment by synthnassizer** [comment]
done

**Comment by SpotlightKid** [comment]
I tested this and I don't think it works as intended. Even with the `--pickup` option, the channel volume immediately jumps to `-inf` gain, when I send a CC event with value 0. It should stay at the initial volume (0.0 dB) until it receives a CC event with value 127 once.

**Comment by synthnassizer** [comment]
I had a quick look at the code and tried to test the functionality by enabling debug output, but I failed to see the volume change messages that I expected. I see  some recurring output from interpolate function, but the updated level due to midi cc messages are no where to be found. I will need to spend some more time in it in the following days when I have some free time.
From what you describe it seems that either the pickup feature does not work correctly or that the order of channel commands in jack_mix_box.c are not in the right order. Not sure if there is even a 3rd option here. Is that the conclusion you also draw?

**Comment by SpotlightKid** [comment]
Tbh, I haven't looked at the code in detail. I saw what you added but apart from that I only tested the behaviour. All I can say is that pickup mode works in the UI version, but not with `jack_mix_box`. IIUC, in `jack_mix_box` it only matters for the for first message of a specific CC anyway, since their is no other way to set the volume than MIDI, so there can only be a discrepancy until the first MIDI CC message for a channel is received. After that, CC messages for that channel should apply directly.

Maybe there's a mis-understanding how the C API is supposed to be used? I didn't implement that, if in doubt, @dsheeler should be able to shed some light on it.

File: jack_mix_box.c [hunk 1]
```diff
@@ -78,9 +78,9 @@ main(int argc, char *argv[])
 		static struct option long_options[] =
 		{
 			{"name",  required_argument, 0, 'n'},
-			{"help",  required_argument, 0, 'h'},
-			{"stereo",  required_argument, 0, 's'},
-			{"pickup",  required_argument, 0, 'p'},
+			{"help",  no_argument, 0, 'h'},
+			{"stereo",  no_argument, 0, 's'},
+			{"pickup",  no_argument, 0, 'p'},
```
Comment (SpotlightKid) [inline jack_mix_box.c:83]: Please keep long options in alphabetical order here.

File: jack_mix_box.c [hunk 2]
```diff
@@ -45,7 +45,7 @@ usage()
 
 	printf("Usage:\n");
 	printf("\tjack_mix_box [ -n|--name <jack client name> ] [ -s|--stereo ] [ -p|--pickup ] [ -v|--volume <initial vol> ] MIDI_CC_1 MIDI_CC_2 ...\n");
-	printf("\tsend SIGUSR1 to the process to have the current columes reported per input channel\n\n");
+	printf("\tsend SIGUSR1 to the process to have the current volumes reported per input channel\n\n");
```
Comment (SpotlightKid) [inline jack_mix_box.c:48]: I reworked the `jack_mix_box` usage message in a commit to the main branch. Please rebase your branch and change the usage message to include a description of the `-p` option.

E.g.:

```diff
diff --git a/jack_mix_box.c b/jack_mix_box.c
index 7a45941..ca5d474 100644
--- a/jack_mix_box.c
+++ b/jack_mix_box.c
@@ -43,10 +43,11 @@ void
 usage()
 {
        printf("Usage: ");
-       printf("jack_mix_box [-n <name>] [-s] [-v <dB>] MIDI_CC...\n");
+       printf("jack_mix_box [-n <name>] [-p] [-s] [-v <dB>] MIDI_CC...\n");
        printf("\n");
        printf("-h|--help\tprint this help message\n");
        printf("-n|--name\tset JACK client name\n");
+       printf("-p|--pickup\tenable MIDI pickup mode (default: jump-to-value)\n");
        printf("-s|--stereo\tmake all input channels stereo with left+right input\n");
        printf("-v|--volume\tinitial volume gain in dB, default is 0.0 (i.e. unity gain)\n");
        printf("\n");
```

**Review by SpotlightKid — CHANGES_REQUESTED** [review]

File: jack_mix_box.c (hunk 1 above)
Comment (synthnassizer) [reply jack_mix_box.c:83]: ok

File: jack_mix_box.c (hunk 2 above)
Comment (synthnassizer) [reply jack_mix_box.c:48]: ok

**Comment by synthnassizer** [comment]
I will close this PR because I managed to make the history look chaotic due to some wrong rebasing that I did.
I have incorporated all the review comments in the new branch that I created.

**Pull request closed without merging**